src/offline/mod_offline_api.erl

```diff
@@ -0,0 +1,42 @@
+-module(mod_offline_api).
+
+-export([delete_expired_messages/1, delete_old_messages/2]).
+
+-spec delete_expired_messages(jid:lserver()) ->
+    {ok | domain_not_found | server_error | module_not_loaded_error, iolist()}.
+delete_expired_messages(Domain) ->
+    check_module_loaded(Domain, fun remove_expired_messages/2, {Domain}).
```

Comment:
It's a bit unexpected that a function called `check_module_loaded` is also running an arbitrary function. Something like `call_for_loaded_module(Domain, Module, Function)` would be better.